tensorflow_quantum/core/ops/tfq_adj_grad_op_test.py

```diff
@@ -0,0 +1,338 @@
+            util.convert_to_tensor([cirq.Circuit()]),
+            tf.convert_to_tensor([], dtype=tf.dtypes.string),
+            tf.convert_to_tensor([[]]),
+            tf.convert_to_tensor([[]], dtype=tf.dtypes.string),
+            tf.convert_to_tensor([[]]))
+        self.assertShapeEqual(np.zeros((1, 0)), out)
+
+    def test_calculate_adj_grad_simple_case(self):
+        """Make sure that adjoint gradient works on simple input case."""
+        n_qubits = 2
+        batch_size = 1
+        symbol_names = ['alpha', 'beta']
+        qubits = cirq.GridQubit.rect(1, n_qubits)
+        circuit_batch, resolver_batch = \
+        [cirq.Circuit(cirq.X(qubits[0]) ** sympy.Symbol('alpha'),
+            cirq.Y(qubits[1]) ** sympy.Symbol('beta'),
+            cirq.CNOT(qubits[0], qubits[1]))], [{'alpha': 0.123, 'beta': 0.456}]
+
+        op_batch = [
+            [cirq.Z(qubits[0]), cirq.X(qubits[1])] for _ in range(batch_size)
+        ]
+
+        symbol_values_array = np.array(
+            [[resolver[symbol]
+              for symbol in symbol_names]
+             for resolver in resolver_batch])
+
+        prev_grads = tf.ones([batch_size, len(symbol_names)])
+
+        out = tfq_adj_grad_op.tfq_adj_grad(
+            util.convert_to_tensor(circuit_batch),
+            tf.convert_to_tensor(symbol_names),
+            tf.convert_to_tensor(symbol_values_array),
+            util.convert_to_tensor(op_batch), prev_grads)
+
+        self.assertAllClose(out, np.array([[-1.18392, 0.43281]]), atol=1e-3)
+
+    def test_calculate_adj_grad_simple_case2(self):
+        """Make sure the adjoint gradient works on another simple input case."""
+        n_qubits = 2
+        batch_size = 1
+        symbol_names = ['alpha', 'beta', 'gamma']
+        qubits = cirq.GridQubit.rect(1, n_qubits)
+        circuit_batch, resolver_batch = \
+        [cirq.Circuit(cirq.X(qubits[0]) ** sympy.Symbol('alpha'),
+            cirq.Y(qubits[1]) ** sympy.Symbol('beta'),
+            cirq.CNOT(qubits[0], qubits[1]),
+            cirq.FSimGate(sympy.Symbol('gamma'), 0.5)(qubits[0], qubits[1]))
+        ], [{'alpha': 0.123, 'beta': 0.456, 'gamma': 0.789}]
+
+        op_batch = [
+            [cirq.Z(qubits[0]), cirq.X(qubits[1])] for _ in range(batch_size)
+        ]
+
+        symbol_values_array = np.array(
+            [[resolver[symbol]
+              for symbol in symbol_names]
+             for resolver in resolver_batch])
+
+        prev_grads = tf.ones([batch_size, len(op_batch[0])])
+
+        out = tfq_adj_grad_op.tfq_adj_grad(
+            util.convert_to_tensor(circuit_batch),
+            tf.convert_to_tensor(symbol_names),
+            tf.convert_to_tensor(symbol_values_array),
+            util.convert_to_tensor(op_batch), prev_grads)
+
+        self.assertAllClose(out,
+                            np.array([[-2.100, -1.7412, -1.5120]]),
+                            atol=1e-3)
+
+    def test_calculate_adj_grad_simple_case3(self):
```

Comment:
I saw that there is no testing case like having one symbol with multiple parameterized gates. Could you please add the case?
```python
        n_qubits = 2
        batch_size = 1
        symbol_names = ['alpha']
        qubits = cirq.GridQubit.rect(1, n_qubits)
circuit_batch, resolver_batch = \
        [cirq.Circuit(cirq.X(qubits[0]) ** sympy.Symbol('alpha'),
            cirq.Y(qubits[1]) ** sympy.Symbol('alpha'),
            cirq.CNOT(qubits[0], qubits[1]),
            cirq.FSimGate(
                sympy.Symbol('alpha'),
                sympy.Symbol('alpha'))(qubits[0], qubits[1]))
        ], [{'alpha': 0.123, 'beta': 0.456, 'gamma': 0.789}]
```

Reply:
Ahhh yes, will add. Thank you.